servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyHttpServer.java

```diff
@@ -385,6 +418,30 @@ public void onComplete() {
             }
         }
 
+        private static void encodePayloadContentIfAvailable(final HttpHeaders requestHeaders,
+                                                            final Set<ContentCoding> supportedEncodings,
+                                                            final StreamingHttpResponse response,
+                                                            final BufferAllocator allocator) {
+            if (supportedEncodings.isEmpty()) {
+                return;
+            }
+
+            ContentCodec codec;
+            if (response.encoding() != null) {
+                codec = response.encoding().codec();
```

Comment:
Considere storing `response.encoding()` result in a local variable to avoid NPE warning for `codec()` method.

You can also move `ContentCodec` variable inside `if`.